pkg/imagestream/imagestream.go

```diff
@@ -0,0 +1,190 @@
+package imagestream
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	imagev1 "github.com/openshift/api/image/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides a struct for imageStream object from the cluster and a imageStream definition.
+type Builder struct {
+	// imageStream definition, used to create the imageStream object.
+	Definition *imagev1.ImageStream
+	// Created imageStream object.
+	Object *imagev1.ImageStream
+	// api client to interact with the cluster.
+	apiClient goclient.Client
+	// Used in functions that define or mutate clusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Pull retrieves an existing imageStream object from the cluster.
+func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("imageStream 'apiClient' cannot be empty")
+	}
+
+	err := apiClient.AttachScheme(imagev1.AddToScheme)
+	if err != nil {
+		glog.V(100).Info("Failed to add image v1 scheme to client schemes")
+
+		return nil, err
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the imageStream is empty")
+
+		return nil, fmt.Errorf("imageStream 'name' cannot be empty")
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the imageStream is empty")
+
+		return nil, fmt.Errorf("imageStream 'nsname' cannot be empty")
+	}
+
+	glog.V(100).Infof(
+		"Pulling imageStream object name %s from namespace %s", name, nsname)
+
+	builder := Builder{
+		apiClient: apiClient.Client,
+		Definition: &imagev1.ImageStream{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+			Spec: imagev1.ImageStreamSpec{},
+		},
+	}
```

Comment:
and it's more just for consistency but could you please move the builder definition to before checking name and nsname parameters?